hack/check-rec-rule-usage.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+set -e
+
+TMP=$(mktemp -d)
+echo "Created temporary directory at $TMP"
+
+# for every rule file
+find assets/ -type f -name "*prometheus-rule.yaml" | while read f
+do
+    # extract recording rules
+    cat $f | tmp/bin/gojsontoyaml -yamltojson | jq '.. | objects | select(.record) | .record' | tr -d '"' | sort -u >> $TMP/rec-rules
+    # and alerting rules
+    cat $f | tmp/bin/gojsontoyaml -yamltojson | jq '.. | objects | select(.alert) | .expr' | grep "\w\+{" -o | tr -d "{" | sort -u >> $TMP/alert-rules
+done
```

Comment:
nit: Since this is an extended version of what we have in https://github.com/openshift/cluster-monitoring-operator/blob/master/Makefile#L129-L132 should we somehow unify both and not duplicate code?

Reply:
Good idea, done.
One question though: https://github.com/openshift/cluster-monitoring-operator/blob/master/Makefile#L129-L132 only gathers the first rule group per file (via `jq -s '[.[]] | { groups: map(.groups[0]) }'`). Is that intentional?
I removed it here and pull in all groups, so this might have unexpected effects.

Reply:
Nevermind, seems this was just fixed in 7f212dfd7f7b8b4e34b2fc6c64892e435b242d55 :)